api/tokens.go

```diff
@@ -0,0 +1,179 @@
+		invalidPasswordMessage := "Unable to validate the password in the request"
+		invalidEmailError := errors.New("Invalid email")
+		invalidErrorMessage := "Unable to validate the email in the request"
+
+		invalidPasswordErrorBody := individualErrorBuilder(invalidPasswordError, invalidPasswordMessage, field, param)
+		invalidEmailErrorBody := individualErrorBuilder(invalidEmailError, invalidErrorMessage, field, param)
+
+		var errorList []IndividualError
+
+		if !(validPasswordRequest) && !(validEmailRequest) {
+
+			errorList = nil
+			errorList = append(errorList, invalidPasswordErrorBody)
+			errorList = append(errorList, invalidEmailErrorBody)
+
+			errorResponseBody := errorResponseBodyBuilder(errorList)
+			writeErrorResponse(ctx, w, 400, errorResponseBody)
+			return
+		}
+
+		if !validPasswordRequest {
+
+			errorList = nil
+			errorList = append(errorList, invalidPasswordErrorBody)
+			errorResponseBody := errorResponseBodyBuilder(errorList)
+
+			writeErrorResponse(ctx, w, 400, errorResponseBody)
+			return
+		}
+
+		if !validEmailRequest {
+
+			errorList = nil
+			errorList = append(errorList, invalidEmailErrorBody)
+			errorResponseBody := errorResponseBodyBuilder(errorList)
+
+			writeErrorResponse(ctx, w, 400, errorResponseBody)
+			return
+		}
+	}
+}
+
+func passwordValidation(requestBody map[string]string) (isPasswordValid bool) {
+
+	isPasswordValid = false
+
+	if len(requestBody["password"]) != 0 {
+		isPasswordValid = true
+	}
+
+	return isPasswordValid
+}
+
+//emailValidation checks for both a valid email address and an empty email address
+func emailValidation(requestBody map[string]string) (isEmailValid bool) {
+
+	isEmailValid = false
+	isEmailValid = emailvalidation.IsEmailValid(requestBody["email"])
+
+	return isEmailValid
+}
+
+func individualErrorBuilder(err error, message, sourceField, sourceParam string) (individualError IndividualError) {
+
+	individualError = IndividualError{
+		SpecificError: error.Error(err),
+		Message:       message,
+		Source: Source{
+			Field: sourceField,
+			Param: sourceParam},
+	}
+
+	return individualError
+}
+
+func errorResponseBodyBuilder(listOfErrors []IndividualError) (errorResponseBody ErrorStructure) {
+
+	errorResponseBody = ErrorStructure{
+		Errors: listOfErrors,
+	}
+
+	return errorResponseBody
+}
+
+func writeErrorResponse(ctx context.Context, w http.ResponseWriter, status int, errorResponseBody interface{}) {
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(status)
+
+	jsonResponse, _ := json.Marshal(errorResponseBody)
+	_, err := w.Write(jsonResponse)
+	if err != nil {
+
+		log.Event(ctx, "writing response failed", log.Error(err), log.ERROR)
+		http.Error(w, "Failed to write http response", http.StatusInternalServerError)
+		return
+	}
+
+	return
+}
+
+func handleUnexpectedError(ctx context.Context, w http.ResponseWriter, err error, message, sourceField, sourceParam string) {
+
+	var errorList []IndividualError
+
+	errorList = nil
```

Comment:
Remove this line - see previous comment.

Reply:
This line has been removed